### PR TITLE
fix: replaced empty tuple error types with Infallible

### PR DESCRIPTION
**Pull request by AlexSherbinin**

### CODE REVIEW

File: dhkem/Cargo.toml [hunk 1]
```diff
@@ -19,10 +19,18 @@ rand_core = "0.6.4"
...
+] }
+p256 = { version = "0.13.2", optional = true, default-features = false, features = [
+	"arithmetic",
+] }
+p384 = { version = "0.13.0", optional = true, default-features = false, features = [
+	"arithmetic",
+] }
+p521 = { version = "0.13.3", optional = true, default-features = false, features = [
+	"arithmetic",
+] }
```
Comment (tarcieri) [inline dhkem/Cargo.toml:33]: This seems unrelated

**Comment by tarcieri** [comment]
Seems like a breaking change, but we'll likely need breaking changes to address #53 as well

File: dhkem/Cargo.toml (hunk 1 above)
Comment (AlexSherbinin) [reply dhkem/Cargo.toml:33]: > This seems unrelated

Autoformatting. Should I remove it?

Comment (tarcieri) [reply dhkem/Cargo.toml:33]: Yes (or at least open a separate PR)

**Comment by faern** [comment]
Yes please! This would be great. Any news on getting this merged?

File: dhkem/Cargo.toml (hunk 1 above)
Comment (AlexSherbinin) [reply dhkem/Cargo.toml]: Done

**Comment by tarcieri** [comment]
Looks like you'll need to update how the errors are handled in `x-wing`, since this is a breaking change

**Comment by tarcieri** [comment]
If you rebase it should now run the full `x-wing` test suite when `ml-kem` is changed

**Review by tarcieri — APPROVED** [review]
